resolver/src/main/java/io/netty/resolver/RoundRobinInetAddressResolver.java

```diff
@@ -64,7 +63,7 @@ public void operationComplete(Future<List<InetAddress>> future) throws Exception
                     if (numAddresses > 0) {
                         // if there are multiple addresses: we shall pick one by one
                         // to support the round robin distribution
-                        promise.setSuccess(inetAddresses.get(index.getAndIncrement() % numAddresses));
+                        promise.setSuccess(inetAddresses.get(shiftSize(numAddresses)));
```

Comment:
@Spikhalskiy - I agree with @guidomedina in that the method name is not very clear. can we rename `shiftSize` to `randomIndex` or something more representative of what it is doing?